examples/SSVEP/plot_classify_ssvep_mdm.py

```diff
@@ -0,0 +1,234 @@
+
+SSVEPEXO_URL = 'https://zenodo.org/record/2392979/files/'
+subject, run = 12, 1
+url = '{:s}subject{:02d}_run{:d}_raw.fif'.format(SSVEPEXO_URL,
+                                                 subject, run + 1)
+sign = 'SSVEPEXO'
+key, key_dest = 'MNE_DATASETS_SSVEPEXO_PATH', 'MNE-ssvepexo-data'
+
+# Use MNE _fetch_file to download EEG file
+if get_config(key) is None:
+    set_config(key, os.path.join(os.path.expanduser("~"), "mne_data"))
+path = _get_path(None, key, sign)
+destination = _url_to_local_path(url, os.path.join(path, key_dest))
+os.makedirs(os.path.dirname(destination), exist_ok=True)
+if not os.path.exists(destination):
+    _fetch_file(url, destination, print_destination=False)
+
+# Read data in MNE Raw and numpy format
+raw = Raw(destination, preload=True, verbose='ERROR')
+events = find_events(raw, shortest_event=0, verbose=False)
+raw = raw.pick_types(eeg=True)
+event_id = {'13 Hz': 2, '17 Hz': 4, '21 Hz': 3, 'resting-state': 1}
+sfreq = int(raw.info['sfreq'])
+
+eeg_data = raw.get_data()
+
+###############################################################################
+# Visualization of raw EEG data
+# -----------------------------
+#
+# Plot few seconds of signal from the `Oz` electrode using matplotlib
+
+n_seconds = 2
+time = np.linspace(0, n_seconds, n_seconds * sfreq,
+                   endpoint=False)[np.newaxis, :]
+plt.figure(figsize=(10, 4))
+plt.plot(time.T, eeg_data[np.array(raw.ch_names) == 'Oz', :n_seconds*sfreq].T,
+         color='C0', lw=0.5)
+plt.xlabel("Time (s)")
+_ = plt.ylabel(r"Oz ($\mu$V)")
+
+###############################################################################
+# And of all electrodes:
+
+plt.figure(figsize=(10, 4))
+for ch_idx, ch_name in enumerate(raw.ch_names):
+    plt.plot(time.T, eeg_data[ch_idx, :n_seconds*sfreq].T, lw=0.5,
+             label=ch_name)
+plt.xlabel("Time (s)")
+plt.ylabel(r"EEG ($\mu$V)")
+plt.legend(loc='upper right')
+
+###############################################################################
+# With MNE, it is much easier to visualize the data
+
+raw.plot(duration=n_seconds, start=0, n_channels=8, scalings={'eeg': 4e-2}, 
+         color={'eeg': 'steelblue'})
+
+###############################################################################
+# Extended signals for spatial covariance
+# ---------------------------------------
+# Using the approach proposed by [1]_, the SSVEP signal is extended to include
+# the filtered signals for each stimulation frequency. We define a function to
+# filter the signal:
+
+def _butter_bandpass(signal, lowcut, highcut, fs, order=4):
+    """ Bilateral Butterworth filter for offline filtering """
+    b, a = butter(order, (lowcut, highcut), btype='bandpass', fs=fs)
+    filtered = filtfilt(b, a, signal, axis=-1)
+    return filtered
+
+
+# we stack the filtered signals to build an extended signal
+frequencies = [13., 17., 21.]
+freq_band = 0.1
+ext_signal = np.vstack([_butter_bandpass(eeg_data, fs=sfreq,
```

Comment:
could you use the MNE filter function instead ?

Reply:
Yes, you are right it is better than creating an ad hoc function. I updated the code to use MNE's filter.